src/components/App/App.js

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import TodoList from "../TodoComponents/TodoList/TodoList";
+import TodoForm from "../TodoComponents/TodoForm/TodoForm";
+import "./App.css";
+import activityList from "./activityList";
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.todoList = null;
+
+    try {
+      this.todoList = JSON.parse(localStorage.taskManager);
+    } catch (e) {
+      this.todoList = activityList;
+    }
+
+    this.state = {
+      todos: this.todoList,
+      newTodoInput: "",
+      searchText: ""
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("beforeunload", () => {
+      this.updateLocalStorage();
+    });
+  }
+
+  updateLocalStorage() {
+    window.localStorage.taskManager = JSON.stringify(this.state.todos);
+  }
+
+  changeHandler = e => {
+    this.setState({
+      newTodoInput: e.target.value
+    });
+  };
+
+  addNewTodo = e => {
+    e.preventDefault();
+
+    if (this.state.newTodoInput.trim()) {
+      const newTodo = {
+        task: this.state.newTodoInput,
+        id: Date.now(),
```

Comment:
Id as a date is fine here, but I believe it's something you should try to avoid, because of the real possibility of  ids with equal value. It's probably better to use a library like uuid for this.